gleam_stdlib/src/str.gleam

```diff
@@ -90,6 +93,22 @@ test from_int {
   |> expect:equal(_, "123")
 }
 
+pub external fn parse_int(String) -> Result(Int, ParseError) = "gleam__stdlib" "parse_int";
+
+test parse_int {
+  "123"
+  |> parse_int
+  |> expect:equal(_, Ok(123))
+
+  "-123"
+  |> parse_int
+  |> expect:equal(_, Ok(-123))
+
+  "0123"
+  |> parse_int
+  |> expect:equal(_, Ok(123))
```

Comment:
Can we have some failing cases as well please?

Reply:
Absolutely!